src/query/plan/variable_start_planner.cpp

```diff
@@ -242,26 +281,35 @@ VaryQueryPartMatching::iterator &VaryQueryPartMatching::iterator::operator++() {
   //    * matchings (m1) and (m2)
   //    * optional matchings (o1) and (o2)
   //    * merge matching (g1)
+  //    * filter matching (f1) and (f2)
   // We want to produce parts for:
-  //    * (m1), (o1), (g1)
-  //    * (m1), (o2), (g1)
-  //    * (m2), (o1), (g1)
-  //    * (m2), (o2), (g1)
+  //    * (m1), (o1), (g1), (f1)
+  //    * (m1), (o1), (g1), (f2)
+  //    * (m1), (o2), (g1), (f1)
+  //    * (m1), (o2), (g1), (f2)
+  //    * (m2), (o1), (g1), (f1)
+  //    * (m2), (o1), (g1), (f2)
+  //    * (m2), (o2), (g1), (f1)
+  //    * (m2), (o2), (g1), (f2)
+
   // Create variations by changing the merge part first.
```

Comment:
Filter part first, we need to change this comment

Reply:
It says " Create variations by changing the **merge**"...
Should be **filter**

Reply:
makes sense